### DISCUSSION TITLE
Disable Libraries?

### DISCUSSION THREAD

**Discussion by Maximilian85** [advplyr/audiobookshelf#120] [Ideas]
Does it make sense to disable libraries so that they are not visible in the overview. For one thing in general I would hide the "Main" Libraries in which all subfolders are as a view. 

As an extension it would be possible to enable only certain libraries per user in the user overview. I have a folder only for children's books and the user "Children" should have access only to this folder but not to the others. 

I hope this was not already a topic.

**Comment by advplyr** [comment]
I'm not sure what you mean by "not visible in the overview".

Having the ability to restrict users to certain libraries was the original plan, but it hasn't been brought up yet. I think I'll do this next along with the bookmarks idea.

**Comment by Maximilian85** [comment]
It would be sufficient if you implant the restriction to certain libraries. Then I can hide the "Main" libraries for all users. Since I cannot remove the "Main" I have to hide it or it will not be added automatically under libraries. 

![Unbenannt1](https://user-images.githubusercontent.com/76774860/138340862-d2e28689-1f4e-4d95-a052-3e61e7d9a65e.PNG)

![Unbenannt2](https://user-images.githubusercontent.com/76774860/138340879-e8a5643e-2dcc-4e4a-a924-639874c26686.PNG)

**Comment by advplyr** [comment]
I just pushed this in v1.4.14

Users default to access all libraries. If you toggle off the all libraries permission then you can select which libraries they will be able to access.

If they try to access an audiobook or library via a direct link that they don't have access to, then it will be treated as if it doesn't exist for the user.

![image](https://user-images.githubusercontent.com/67830747/138536581-46a3b6e7-84c3-4cb8-89c2-ef527e245d93.png)

Let me know if I missed something there.

**Comment by Maximilian85** [comment]
Thank you that works, thank you very much.

**Comment by Thomas-Magnum** [comment]
Sorry for digging out this old thread...
I haven't found a way enable or disable libraries for _existing_ users. When setting up a new user I'm able to assign certain libraries for this user. However for existing users there seems to be no way to add or remove access to libraries...

**Reply by Maximilian85** [reply]
I don't have any old users but there should be an option if you are logged in as a root user. 
Edit an existing user with the pencil icon and uncheck "Can Access All Libraries" then a selection should appear below this.

**Reply by Thomas-Magnum** [reply]
Thanks for your reply. 
You're absolutely right - I constantly must have searched for this option in the wrong place!
